### PR TITLE
boards/arm: Remove "LDFLAGS += -g" whichi is already added by Toolchain.defs

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
risc-v: Move "LDFLAGS += -melf32lriscv" from Make.defs to Toolchain.defs

## Impact
Code refactor

## Testing
Pass CI

### CODE REVIEW

**Review by hartmannathan — COMMENTED** [review]
LGTM on code reading. Can't build-test at the moment: Missing a RISC-V toolchain on my system. However AFAICT the -g flag is being used correctly.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

File: boards/arm/c5471/c5471evm/scripts/Make.defs [hunk 1]
```diff
@@ -69,7 +69,3 @@ NXFLATLDFLAGS2 = $(NXFLATLDFLAGS1) \
                  -T$(TOPDIR)/binfmt/libnxflat/gnu-nxflat-gotoff.ld \
                  -no-check-sections
 LDNXFLATFLAGS = -e main -s 2048
-
-ifeq ($(CONFIG_DEBUG_FEATURES),y)
```
Comment (pkarashchenko) [inline boards/arm/c5471/c5471evm/scripts/Make.defs:73]: Let's remove
```
ifeq ("${CONFIG_DEBUG_FEATURES}","y")
  ARCHOPTIMIZATION = -g
endif
```
as well

File: boards/arm/lpc214x/mcu123-lpc214x/scripts/Make.defs [hunk 2]
```diff
@@ -69,7 +69,3 @@ NXFLATLDFLAGS2 = $(NXFLATLDFLAGS1) \
                  -T$(TOPDIR)/binfmt/libnxflat/gnu-nxflat-gotoff.ld \
                  -no-check-sections
 LDNXFLATFLAGS = -e main -s 2048
-
-ifeq ($(CONFIG_DEBUG_FEATURES),y)
```
Comment (pkarashchenko) [inline boards/arm/lpc214x/mcu123-lpc214x/scripts/Make.defs:73]: Let's remove
```
ifeq ("${CONFIG_DEBUG_FEATURES}","y")
  ARCHOPTIMIZATION = -g
endif
```
as well

File: boards/arm/lpc214x/zp214xpa/scripts/Make.defs [hunk 3]
```diff
@@ -69,7 +69,3 @@ NXFLATLDFLAGS2 = $(NXFLATLDFLAGS1) \
                  -T$(TOPDIR)/binfmt/libnxflat/gnu-nxflat-gotoff.ld \
                  -no-check-sections
 LDNXFLATFLAGS = -e main -s 2048
-
-ifeq ($(CONFIG_DEBUG_FEATURES),y)
```
Comment (pkarashchenko) [inline boards/arm/lpc214x/zp214xpa/scripts/Make.defs:73]: Let's remove
```
ifeq ("${CONFIG_DEBUG_FEATURES}","y")
  ARCHOPTIMIZATION = -g
endif
```
as well

File: boards/arm/lpc2378/olimex-lpc2378/scripts/Make.defs [hunk 4]
```diff
@@ -69,7 +69,3 @@ NXFLATLDFLAGS2 = $(NXFLATLDFLAGS1) \
                  -T$(TOPDIR)/binfmt/libnxflat/gnu-nxflat-gotoff.ld \
                  -no-check-sections
 LDNXFLATFLAGS = -e main -s 2048
-
-ifeq ($(CONFIG_DEBUG_FEATURES),y)
```
Comment (pkarashchenko) [inline boards/arm/lpc2378/olimex-lpc2378/scripts/Make.defs:73]: Let's remove
```
ifeq ("${CONFIG_DEBUG_FEATURES}","y")
  ARCHOPTIMIZATION = -g
endif
```
as well

File: boards/arm/str71x/olimex-strp711/scripts/Make.defs [hunk 5]
```diff
@@ -69,7 +69,3 @@ NXFLATLDFLAGS2 = $(NXFLATLDFLAGS1) \
                  -T$(TOPDIR)/binfmt/libnxflat/gnu-nxflat-gotoff.ld \
                  -no-check-sections
 LDNXFLATFLAGS = -e main -s 2048
-
-ifeq ($(CONFIG_DEBUG_FEATURES),y)
```
Comment (pkarashchenko) [inline boards/arm/str71x/olimex-strp711/scripts/Make.defs:73]: Let's remove
```
ifeq ("${CONFIG_DEBUG_FEATURES}","y")
  ARCHOPTIMIZATION = -g
endif
```
as well

File: boards/arm/c5471/c5471evm/scripts/Make.defs (hunk 1 above)
Comment (xiaoxiang781216) [reply boards/arm/c5471/c5471evm/scripts/Make.defs:71]: Done.

File: boards/arm/lpc214x/zp214xpa/scripts/Make.defs (hunk 3 above)
Comment (xiaoxiang781216) [reply boards/arm/lpc214x/zp214xpa/scripts/Make.defs:71]: Done.

File: boards/arm/lpc214x/mcu123-lpc214x/scripts/Make.defs (hunk 2 above)
Comment (xiaoxiang781216) [reply boards/arm/lpc214x/mcu123-lpc214x/scripts/Make.defs:71]: Done.

File: boards/arm/lpc2378/olimex-lpc2378/scripts/Make.defs (hunk 4 above)
Comment (xiaoxiang781216) [reply boards/arm/lpc2378/olimex-lpc2378/scripts/Make.defs:71]: Done.

File: boards/arm/str71x/olimex-strp711/scripts/Make.defs (hunk 5 above)
Comment (xiaoxiang781216) [reply boards/arm/str71x/olimex-strp711/scripts/Make.defs:71]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Review by Ouss4 — APPROVED** [review]

**Pull request merged**